assets/js/base/context/providers/cart-checkout/checkout-processor.ts

```diff
@@ -275,7 +276,8 @@ const CheckoutProcessor = () => {
 						)
 						.then( ( response: CheckoutResponseError ) => {
 							if ( response.data?.cart ) {
-								receiveCart( response.data.cart );
+								// We don't want to receive the address here because it will overwrite fields.
+								receiveCartContents( response.data.cart );
```

Comment:
This fixes the address data being lost after a failed checkout request if data has not yet persisted. https://a8c.slack.com/archives/C8X6Q7XQU/p1674643137575739

Reply:
I couldn't find where `receiveCartContents` is defined so I assume it was already in trunk, I'm trying to understand its role here, at least, I understand that it sets everything except the addresses, but why do we need to do that? Why is the data returned from the server inconsistent? if addresses didn't persist, shouldn't we assume line items and such are going to be incorrect here and just reject the whole thing and show errors?

Reply:
It's in trunk already (https://github.com/woocommerce/woocommerce-blocks/pull/7711). Its there because when you're persisting address data to the server, in particular partial address data, it is not desirable to update the stores with the response—the data in the stores may be more current. We needed this change after introducing partial pushes, since the partial push does not include all current data.

Reply:
Yes and when editing (`updateCustomerData`), which is where `receiveCartContents` was already in use.